lib/cybersource/api.rb

```diff
@@ -336,6 +352,22 @@ def add_required_options(kb_account_id, properties, options, context)
         ::Killbill::Plugin::ActiveMerchant::Utils.normalize_property(properties, 'ignore_cvv')
       end
 
+      def add_merchant_descriptor(transaction_type, properties, options)
+        merchant_descriptor = find_value_from_properties(properties, 'merchant_descriptor')
+        return unless merchant_descriptor.present?
+        
+        merchant_descriptor_hash = JSON.parse(merchant_descriptor) rescue nil
+        return unless merchant_descriptor_hash.present? && merchant_descriptor_hash.is_a?(Hash)
+
+        merchant_descriptor_hash = merchant_descriptor_hash.with_indifferent_access
+
+        merchant_descriptor_hash[:transaction_type] = transaction_type
+        if merchant_descriptor_hash[:card_type].nil?
+          merchant_descriptor_hash[:card_type] = find_value_from_properties(properties, 'cc_type')
```

Comment:
Hi @pierre , is this the correct way of retrieve `cc_type` from properties?
This is the first place to get cc_type from input, and in the `before_gateways` will be the last resort to populate this value.